libethash-cuda/CUDAMiner.cpp

```diff
@@ -454,65 +454,66 @@ void CUDAMiner::search(
         {
             cudaStream_t stream = m_streams[current_index];
             buffer = m_search_buf[current_index];
-            // Wait for stream batch to complete
+
+            // Wait for stream batch to complete and immediately
+            // store number of processed hashes
             CUDA_SAFE_CALL(cudaStreamSynchronize(stream));
+            addHashCount(batch_size);
```

Comment:
Why increment hashcount at the to instead of the bottom of the loop? What's the difference?

Reply:
Only a "logic" reason, not a technical one.
When cuda stream sync returns the batch is complete. No need to postpone later in the code the collection of the value: it's not affected by operations.

Reply:
Ah I see. This is good! We were previously not counting the hashes from the first batch for each job.